framework/packages/abstract-interface/src/account/mod.rs

```diff
@@ -285,6 +291,119 @@ impl<Chain: CwEnv> AbstractAccount<Chain> {
         Ok(migrated)
     }
 
+    pub fn upgrade(&self) -> Result<(), AbstractInterfaceError> {
+        let mut one_migration_was_successful = false;
+
+        // We upgrade the manager to the latest version through all the versions
+        loop {
+            if self.upgrade_next_module_version(MANAGER).is_err() {
+                break;
+            }
+            one_migration_was_successful = true;
+        }
+
+        loop {
+            if self.upgrade_next_module_version(PROXY).is_err() {
+                break;
+            }
+            one_migration_was_successful = true;
+        }
```

Comment:
Not sure if they should be updated separately. But it's possible versions won't be the same in the future(hopefully we will merge them into one contract sooner)

Reply:
For now its fine but we should either do migrations in pairs or have them merged in the future as you say. One way to do that is to block migrations for proxy and do the proxy upgrade along with a manager upgrade when requested.

Reply:
When requested by what ? Is there a dependency mechanism you are thinking of ?